samples/boards/olimex_stm32_e407/ccm/README.rst

```diff
@@ -0,0 +1,186 @@
+.. _olimex_stm32_e407_ccm:
+
+Olimex STM32-E407 CCM example
+#############################
+
+Overview
+********
+
+Show usage of the Core Coupled Memory (CCM) that is available
+on several STM32 devices. The very important difference with
+normal RAM is that CCM can not be used for DMA.
+
+By prefixing a variable with __ccm_data_section, __ccm_bss_section,
+or __ccm_noinit_section those variables are placed in the CCM.
+
+The __ccm_data_section prefix should be used for variables that
+are initialized. Like the normal data section the initial
+values take up space in the FLASH image.
+
+The __ccm_bss_section prefix should be used for variables that
+should be initialized to 0. Like the normal bss section they
+do not take up FLASH space.
+
+The __ccm_noinit_section prefix should be used for variables
+that don't need to have a defined initial value (for example
+buffers that will receive data). Compared to bss or data the
+kernel does not need to initialize the noinit section making
+the startup slightly faster.
+
+To add CCM support to a board the following line will have to
+be added the the boards DTS file chosen section;
+
+.. code-block:: console
+
+   zephyr,ccm = &ccm0;
+
+For example the olimex STM32 E407 DTS file looks like this;
+
+.. code-block:: console
+
+   /*
+    * Copyright (c) 2017, Erwin Rol <erwin@erwinrol.com>
+    *
+    * SPDX-License-Identifier: Apache-2.0
+    */
+
+   /dts-v1/;
+   #include <st/stm32f407.dtsi>
+
+   / {
+           model = "Olimex STM32-E407 board";
+           compatible = "olimex,stm32-e407", "st,stm32f407";
+
+           chosen {
+                   zephyr,console = &usart1;
+                   zephyr,sram = &sram0;
+                   zephyr,flash = &flash0;
+                   zephyr,ccm = &ccm0;
+           };
+   };
+
+   &usart1 {
+           current-speed = <115200>;
+           pinctrl-0 = <&usart1_pins_a>;
+           pinctrl-names = "default";
+           status = "ok";
+   };
+
+   &usart3 {
+           current-speed = <115200>;
+           pinctrl-0 = <&usart3_pins_a>;
+           pinctrl-names = "default";
+           status = "ok";
+   };
+
+   &usart6 {
+           current-speed = <115200>;
+           pinctrl-0 = <&usart6_pins_b>;
+           pinctrl-names = "default";
+           status = "ok";
+   };
+
+
+Building
+********
+
+.. code-block:: console
+
+   $ cd samples/boards/olimex_stm32_e407/ccm
+   $ make
+
+Running
+*******
+
+The first time the example is ran after power on the output will look
+someting like this;
+
+.. code-block:: console
+
+   ***** BOOTING ZEPHYR OS v1.11.0 *****
```

Comment:
We're not at version 1.11 yet, should this be 1.10.99 for now?

Reply:
@dbkinder  Thank you for taking the time to proofread and the suggestions, I pretty much implemented them 1 to 1. But with the version number , when I now write 1.10.99 it will never be changed again , will it? Wouldn't it than be better to use the version number of the upcoming release (1.11.0 in this case) ?

Reply:
Well, at least if you run it now, it will show 1.10.99 (if using the master branch) until the 1.11.0 release comes out in March 2018 and then 1.12.0 in June (or so)...

Reply:
So just make it 1.10.99 ?

Reply:
Yes (my recommendation).